Sources/PostgreSQL/Database/PostgreSQLDatabaseConfig.swift

```diff
@@ -23,9 +23,9 @@ public struct PostgreSQLDatabaseConfig {
     public let password: String?
     
     /// Creates a new `PostgreSQLDatabaseConfig`.
-    public init(hostname: String, port: Int = 5432, username: String, database: String? = nil, password: String? = nil) {
+    public init(hostname: String, port: Int?, username: String, database: String? = nil, password: String? = nil) {
         self.hostname = hostname
-        self.port = port
+        self.port = port ?? 5432
```

Comment:
What's the reasoning behind this change? `port` being Optional doesn't make a whole lot of sense in my opinion because it doesn't accurately reflect the fact that a port is required.

Reply:
Since URLs allow for no port to be specified, I was motivated to keep parity across all initializers.
On second thought changing the signature will require changes elsewhere, outside the scope of what this PR is supposed to address. Will revert this bit.

Reply:
> Since URLs allow for no port to be specified
I didn't know that! Very interesting. 

Does that mean clients are expected to be able to connect to Postgres without a port specified in that case?

Reply:
https://www.postgresql.org/docs/current/static/libpq-connect.html#LIBPQ-CONNSTRING
Section `33.1.1.2. Connection URIs` seems to indicate so. Unfortunately there is no access to username from the environment otherwise `postgresql://` would have been valid as well.